tools/tvk-oneclick/tvk-oneclick.sh

```diff
@@ -738,14 +768,24 @@ create_aws_s3() {
     read -r -p "Bucket Name: " bucket_name
     read -r -p "Target Name: " target_name
     read -r -p "Target Namespace: " target_namespace
-    read -r -p "thresholdCapacity(Units can be[Mi/Gi/Ti]) (default - 1000Gi): " thresholdCapacity
+    read -r -p "thresholdCapacity (Units can be[Mi/Gi/Ti]) (default - 1000Gi): " thresholdCapacity
     read -r -p "Proceed even if resource exists y/n (default - y): " if_resource_exists_still_proceed
   fi
   if [[ -z "$if_resource_exists_still_proceed" ]]; then
     if_resource_exists_still_proceed='y'
   fi
+  if [[ -z "$target_namespace" ]]; then
+    target_namespace="default"
+  fi
   if [[ $(kubectl get target "$target_name" -n "$target_namespace" 2>> >(logit)) ]]; then
-    echo "Target with same name already exists"
+    if kubectl get target "$target_name" -n "$target_namespace" -o 'jsonpath={.status.status}' 2>/dev/null | grep -q Unavailable; then
+      echo "Target with same name already exists but is in Unavailable state"
+      return 1
+    else
+      echo "Target with same name already exists"
+      return 0
```

Comment:
For target status `InProgress`, it'll return 0.. would it be okay?

Reply:
It is handled. The function which calls this takes care of it.
If its InProgress in create_target() function I have wait for some time then aborted.